packages/react-components/react-tree/src/components/Tree/useTree.ts

```diff
@@ -12,13 +14,94 @@ import type { TreeProps, TreeState } from './Tree.types';
  * @param ref - reference to root HTMLElement of Tree
  */
 export const useTree_unstable = (props: TreeProps, ref: React.Ref<HTMLElement>): TreeState => {
+  const {
+    isSubtree,
+    level: parentLevel,
+    openTrees: rootOpenTrees,
+    requestOpenChange: rootRequestOpenChange,
+    treeRef: parentTreeRef,
+  } = useTreeContext_unstable();
+  warnIfNecessary(props, isSubtree);
+  const { openSubtrees: stateOpenTrees, defaultOpenSubtrees: defaultOpenTrees, onOpenChange, ...rest } = props;
+  const [localOpenTrees, setOpenTrees] = useControllableState({
```

Comment:
each Tree having a state doesn't quite make sense to me. Would it make more sense to have some check here so that only the root tree uses state and subtrees would simply read from context to check if they are open or not. `setOpen` can be passed down through context too

Reply:
I'm actually just ignoring local state whenever it's a subtree:

https://github.com/microsoft/fluentui/blob/c626b503d41e07234f74b98326ccabc2b01b7df9/packages/react-components/react-tree/src/components/Tree/useTree.ts#L46-L47


you're suggesting for using the hooks inside a conditional?